mypy/nodes.py

```diff
@@ -174,10 +176,10 @@ class SymbolNode(Node):
     # TODO do not use methods for these
 
     @abstractmethod
-    def name(self) -> str: pass
+    def name(self) -> Bogus[str]: pass
```

Comment:
I would add a comment here (the place `Bogus` first appears in this file) explaining what is going on. My understanding is that there are quite a few places where `name` or `fullname` can be actually `None` while this is not reflected in the type.

Btw, I am surprised you need this for `name` too. I have seen several bugs because of `fullname` being `None`, but I have never encountered a situation where `name` is `None`. Did you see actual problems with it, or you just decided this from analogy with `fullname`?

Reply:
I saw actual problems, but they can be easily fixed by setting a default _name.

Reply:
To the empty string, that is. `MypyFile` appears to never have a real name set?

Reply:
I would prefer something like `id.split('.')[-1]`, if it is possible of course.

Reply:
(for `MypyFile`s)